samplers/aws/xray/internal/reservoir.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
Considering copying the code in from here to not have sampling decisions aligned on epoch second boundaries like this implementation. The alignment has some potential for thundering herd problems so is somewhat operationally unsafe

https://github.com/jaegertracing/jaeger-client-go/blob/master/utils/rate_limiter.go